src/handler.ts

```diff
@@ -262,10 +312,31 @@ export class DebuggerHandler {
   private _type: DebuggerHandler.SessionType;
   private _shell: JupyterFrontEnd.IShell;
   private _service: IDebugger;
+  private _previousConnection: Session.ISessionConnection;
```

Comment:
Not sure having this `_previousConnection` is the way to go, as it adds another piece of information to keep track of.

Maybe an alternative would be to have something like `service.isAvailable()` but to ask whether the debug session has started? (using the `debugInfo` request).